### PR TITLE
importer: auto split when import end

### PR DESCRIPTION
**Pull request by Little-Wallace**
Signed-off-by: Little-Wallace <bupt2013211450@gmail.com>

### What problem does this PR solve?

Problem Summary:
If region split during import, the sst_importer::write rpc has to retry and it may hurt performance of import. And lightning may import a lot of data to one region, this may cause a large region. So I hope TiKV can split the region when import end.

### What is changed and how it works?

TiKV will check the import mode of importer before scheduling a split-check task.  If the approximate size is too large, TiKV will schedule a split-check task.

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- PR to update `pingcap/tidb-ansible`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Manual test (add detailed scripts or steps below)

Side effects

- Performance regression
    - Consumes more CPU.  There may be a lot of split-check task after import task ending, and these tasks will cost more CPU.

### Release note 
- No release note.

### CODE REVIEW

File: components/raftstore/src/store/fsm/peer.rs [hunk 1]
```diff
@@ -3546,6 +3560,10 @@ where
             return;
         }
 
+        if self.ctx.importer.get_mode() == SwitchMode::Import {
```
Comment (overvenus) [inline components/raftstore/src/store/fsm/peer.rs:3563]: Please add some comments about skipping the split check.

File: components/raftstore/src/store/fsm/peer.rs [hunk 2]
```diff
@@ -3584,7 +3607,7 @@ where
         }
         self.fsm.peer.size_diff_hint = 0;
         self.fsm.peer.compaction_declined_bytes = 0;
-        self.register_split_region_check_tick();
+        return true;
```
Comment (overvenus) [inline components/raftstore/src/store/fsm/peer.rs:3610]:
```suggestion
       true
```

File: components/raftstore/src/store/fsm/peer.rs [hunk 3]
```diff
@@ -3560,6 +3575,13 @@ where
             return;
         }
 
+        // When Lightning or BR is importing data to TiKV, their ingest-request may fail because of
+        // region-epoch not matched. So we hope TiKV do not check region size and split region during
+        // importing.
+        if self.ctx.importer.get_mode() == SwitchMode::Import {
```
Comment (Connor1996) [inline components/raftstore/src/store/fsm/peer.rs:3581]: Move it to the line between L3595 and L3596. For most cases, the check will be returned at the check of size_diff_hint, moving the import mode check can save unnecessary atomic load operations which involves memory fence.

File: components/raftstore/src/store/fsm/peer.rs [hunk 4]
```diff
@@ -523,6 +524,20 @@ where
...
+                PeerMsg::SplitCheck => {
+                    // If the current `approximate_size` is larger than `region_max_size`, this region
+                    // may ingest a large file which is imported by `BR` or `lightning`, we shall check it
+                    let approximate_size =
+                        self.fsm.peer.approximate_size.unwrap_or(u64::max_value());
+                    let approximate_keys =
+                        self.fsm.peer.approximate_keys.unwrap_or(u64::max_value());
+                    if self.fsm.peer.is_leader()
+                        && (approximate_size > self.ctx.coprocessor_host.cfg.region_max_size.0
+                            || approximate_keys > self.ctx.coprocessor_host.cfg.region_max_keys)
```
Comment (Connor1996) [inline components/raftstore/src/store/fsm/peer.rs:536]:
```suggestion
                    if self.fsm.peer.is_leader()
                        && (self.fsm.peer.approximate_size.map_or(true, |v| v > self.ctx.coprocessor_host.cfg.region_max_size.0) 
                            || self.fsm.peer.approximate_keys.map_or(true, |v| v > self.ctx.coprocessor_host.cfg.region_max_keys))
                        && (approximate_size > self.ctx.coprocessor_host.cfg.region_max_size.0
                            || approximate_keys > self.ctx.coprocessor_host.cfg.region_max_keys)
```

**Review by Connor1996 — COMMENTED** [review]
rest LGTM

**Review by Connor1996 — APPROVED** [review]

**Pull request merged**